modules/javafx.controls/src/main/java/javafx/scene/control/skin/TabPaneSkin.java

```diff
@@ -991,8 +991,10 @@ private void removeTab(Tab tab) {
         }
 
         private void moveTab(int moveToIndex, TabHeaderSkin tabHeaderSkin) {
-            headersRegion.getChildren().remove(tabHeaderSkin);
-            headersRegion.getChildren().add(moveToIndex, tabHeaderSkin);
+            if (moveToIndex != headersRegion.getChildren().indexOf(tabHeaderSkin)) {
```

Comment:
Unless I am missing something, this check seems unrelated to the bug.

Reply:
Yes Kevin, this is not required for fix. In some scenarios the tabHeader gets placed at correct index due to the addition of new headers, so this movement becomes redundant.

Reply:
In that case, this change is fine.